TZMBot/cogs/sar.py

```
@@ -15,39 +15,52 @@ def __init__(self, client):
 
         self.dict = {
             emoji: role_id
-            for category in self.config["categories"].values()
+            for message_id in self.config.keys()
+            for category in self.config[message_id]["categories"].values()
             for emoji, role_id in category.items()
         }
 
         self.channel = None
         self.message = None
         self.active = False
-        self.client.loop.create_task(self.async_setup())
+        self.messages = {}
+        self.client.loop.create_task(self.async_call_setup())
 
-    async def async_setup(self):
+    async def async_call_setup(self):
         await self.client.wait_until_ready()
 
+        try:
+            await self.async_setup()
+        except Exception as e:
+            logger.error("SelfAssignableRoles ERROR " + str(e))
```

do we want to reraise the exceptions?